src/admin/metrics.rs

```diff
@@ -0,0 +1,336 @@
+use http_body_util::Full;
+use hyper::{
+    body::Bytes,
+    server::conn::http1,
+    service::service_fn,
+    Request,
+};
+use prometheus_metric_storage::{
+    MetricStorage,
+    StorageRegistry,
+};
+use std::convert::Infallible;
+use std::{
+    collections::hash_map::HashMap,
+    sync::{
+        Arc,
+        RwLock,
+    },
+    time::Duration,
+};
+
+use tokio::sync::{
+    mpsc::{
+        unbounded_channel,
+        UnboundedReceiver,
+        UnboundedSender,
+    },
+    oneshot,
+};
+
+use thiserror::Error;
```

Comment:
does thiserror use proc macros? if so we cant use it because it nukes compile times

Reply:
Ah, yeah , 
Prolly could just use tokio::mpsc::error instead if I need to